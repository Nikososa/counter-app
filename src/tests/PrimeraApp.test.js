import '@testing-library/jest-dom';
import React from 'react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Pruebas en PrimeraApp', () => {
    
    test('Debe mostrar PrimeraApp correctamente', () => {

        const saludo = 'Bienvenido';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } />)

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('Debe mostrar el subtitulo correctamente', () => {

        const saludo = 'Bienvenido';
        const subtitulo = 'Prueba de primeraApp'
        const wrapper = shallow( 
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subtitulo }
            />
        )

        const textoParrafo = wrapper.find('p').text(); // Recupera elemento 'p' y obitene el texto que encierra
        
        expect( textoParrafo ).toBe( subtitulo );

    })
})
