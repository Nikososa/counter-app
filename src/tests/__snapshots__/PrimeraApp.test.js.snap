// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe mostrar PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Bienvenido
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
